src/state/migrations/1.ts

```diff
@@ -0,0 +1,91 @@
+import { PersistedState } from 'redux-persist';
+
+import { EvmDerivedAddresses, SolDerivedAddresses } from '@/constants/account';
+import { ConnectorType, EvmAddress, WalletInfo, WalletNetworkType } from '@/constants/wallets';
+
+import { WalletState } from '../wallet';
+import { parseStorageItem } from './utils';
+
+type V1State = PersistedState & { wallet: WalletState };
+/**
+ * Move over wallet data from localStorage into redux
+ * TODO (in future migration): Remove these localStorage items
+ */
+export function migration1(state: PersistedState): V1State {
+  if (!state) {
+    throw new Error('state must be defined');
+  }
+
+  const evmAddress = localStorage.getItem('dydx.EvmAddress') as EvmAddress | undefined;
+  const evmDerivedAddresses = parseStorageItem<EvmDerivedAddresses>(
+    localStorage.getItem('dydx.EvmDerivedAddresses')
+  );
+
+  const solAddress = localStorage.getItem('dydx.SolAddress');
+  const solDerivedAddresses = parseStorageItem<SolDerivedAddresses>(
+    localStorage.getItem('dydx.SolDerivedAddresses')
+  );
+
+  const dydxAddress = localStorage.getItem('dydx.DydxAddress');
+  const selectedWallet = parseStorageItem<WalletInfo>(
+    localStorage.getItem('dydx.OnboardingSelectedWallet')
+  );
+
+  if (!selectedWallet) {
+    return {
+      ...state,
+      wallet: {
+        sourceAccount: {
+          address: undefined,
+          chain: undefined,
+          encryptedSignature: undefined,
+          walletInfo: undefined,
+        },
+      },
+    };
+  }
+
+  if (selectedWallet.connectorType === ConnectorType.PhantomSolana) {
```

Comment:
does TS have exhaustive switching? would be good to ensure all cases of `ConnectorType` are covered if we add a new one.

Reply:
kind of, but i think for this instance, migration files are only run once so it technically should just cover the enum values that exist at this point in time, so if we add a new ConnectorType, that should technically go in a new migration file instead of having to edit this one